scripts/lib/CIME/utils.py

```diff
@@ -26,11 +26,40 @@ def expect(condition, error_msg, exc_type=SystemExit, error_prefix="ERROR:"):
         if logger.isEnabledFor(logging.DEBUG):
             import pdb
             pdb.set_trace()
-        raise exc_type("%s %s" % (error_prefix,error_msg))
+        logger.error("%s %s", error_prefix,error_msg, exc_info=True)
+        raise exc_type("%s %s"%(error_prefix,error_msg))
 
 def id_generator(size=6, chars=string.ascii_lowercase + string.digits):
     return ''.join(random.choice(chars) for _ in range(size))
 
+def check_name(fullname, additional_chars=None):
```

Comment:
Thanks, this looks better. One thing that may be good is to have check_name return a true/false and then let the user decide if an exception is needed.